packages/edit-site/src/components/style-book/index.js

```diff
@@ -316,7 +329,86 @@ function StyleBook( {
 	);
 }
 
-const StyleBookBody = ( {
+/**
+ * Style Book Preview component renders the stylebook without the Editor dependency.
+ *
+ * @param {Object}   props            Component props.
+ * @param {string}   props.path       Path to the selected block.
+ * @param {Object}   props.userConfig User configuration.
+ * @param {Function} props.isSelected Function to check if a block is selected.
+ * @param {Function} props.onSelect   Function to select a block.
+ * @return {Object} Style Book Preview component.
+ */
+export const StyleBookPreview = ( {
+	path = '',
+	userConfig = {},
+	isSelected,
+	onSelect,
+} ) => {
+	const siteEditorSettings = useSelect(
+		( select ) => select( siteEditorStore ).getSettings(),
+		[]
+	);
+	// Update block editor settings because useMultipleOriginColorsAndGradients fetch colours from there.
+	dispatch( blockEditorStore ).updateSettings( siteEditorSettings );
```

Comment:
If we wanted to avoid this we could potentially move this logic inside `BlockEditorProvider` but that would mean adding another layer of component wrapping 😅

Reply:
Is this dispatch still necessary? Looks like hoge already exists: https://github.com/WordPress/gutenberg/pull/66851/files#diff-7b6d15f7afd4674f10f20f7e1a5a21d04891e84996c308dae854d6b97dcfc21cR418

Reply:
Yeah, the problem is that the color palette info is in the site editor settings, but `useMultipleOriginColorsAndGradients` is a block editor function and uses block editor settings, so we need to update those before calling it.